hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/TestLocalProcessLockProvider.java

```diff
@@ -0,0 +1,253 @@
+ * limitations under the License.
+ */
+
+package org.apache.hudi.client.transaction;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hudi.client.transaction.lock.LocalProcessLockProvider;
+import org.apache.hudi.common.config.LockConfiguration;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.exception.HoodieLockException;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class TestLocalProcessLockProvider {
+
+  private static final Logger LOG = LogManager.getLogger(TestLocalProcessLockProvider.class);
+  private final Configuration hadoopConfiguration = new Configuration();
+  private final LockConfiguration lockConfiguration = new LockConfiguration(new TypedProperties());
+
+  @Test
+  public void testLockAcquisition() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.lock();
+    });
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+  }
+
+  @Test
+  public void testLockReAcquisitionBySameThread() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.lock();
+    });
+    assertThrows(HoodieLockException.class, () -> {
+      localProcessLockProvider.lock();
+    });
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+  }
+
+  @Test
+  public void testLockReAcquisitionByDifferentThread() {
+    LocalProcessLockProvider localProcessLockProvider = new LocalProcessLockProvider(lockConfiguration, hadoopConfiguration);
+    final AtomicBoolean writer2Completed = new AtomicBoolean(false);
+
+    // Main test thread
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.lock();
+    });
+
+    // Another writer thread in parallel, should block
+    // and later acquire the lock once it is released
+    Thread writer2 = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        assertDoesNotThrow(() -> {
+          localProcessLockProvider.lock();
+        });
+        assertDoesNotThrow(() -> {
+          localProcessLockProvider.unlock();
+        });
+        writer2Completed.set(true);
+      }
+    });
+    writer2.start();
+
+    assertDoesNotThrow(() -> {
+      localProcessLockProvider.unlock();
+    });
+
+    try {
+      writer2.join();
+    } catch (InterruptedException e) {
+      //
```

Comment:
Let's clean this one up it is not very useful anyway

Reply:
fixed.